test/options.test.ts

```diff
@@ -119,6 +119,39 @@ describe("File extensions", () => {
   });
 });
 
+describe("Headings", () => {
+  it("doesn't add IDs or export headings by default", async () => {
+    const result = (await markdocPreprocess().markup!({
+      content: basicMarkdoc,
+      filename: "test.md",
+    })) as Processed;
+
+    expect(result.code).toMatchSnapshot();
+  });
+
+  it("adds IDs and exports headings when passed as an option", async () => {
+    const result = (await markdocPreprocess({
+      headingIds: true,
+    } as Options).markup!({
+      content: basicMarkdoc,
+      filename: "test.md",
+    })) as Processed;
+
+    expect(result.code).toMatchSnapshot();
+  });
+
+  it("doesn't add IDs or export headings when passed false", async () => {
+    const result = (await markdocPreprocess({
+      headingIds: false,
+    } as Options).markup!({
+      content: basicMarkdoc,
+      filename: "test.md",
+    })) as Processed;
+
+    expect(result.code).toMatchSnapshot();
+  });
+});
```

Comment:
Good idea to test the options!

A test for passing the function `headingIds: (title: string) => string` as an option is already covered by the headings tests too, so that will catch if the behaviour changes in the future.